ruby/sorting-words-and-numbers/main.rb

```diff
@@ -1,16 +1,19 @@
 # frozen_string_literal: true
+SEPARATER = ' '
 
-loop do
-  sorted = [[], []]
-  line = gets
-  next if line.nil?
+while line = gets.chomp
+  classified_words = {
+    numerical_digits: [],
```

Comment:
### Why `numerical_digits` ?

First of all, `number` is a bad name, because it contains integer, real number, and even complex number.
But, this program does not target real numbers or complex numbers.
Therefore, number is not a good name. The same goes for `real_number`.
ref: https://en.wikipedia.org/wiki/Number

What about `integer`?
Integer is also a bad name. It is true that an integer does not contain a decimal point. But an integer contains a negative number. And this program does not target negative numbers.
ref: https://en.wikipedia.org/wiki/Integer

What about `natural_number`? Natural numbers do not contain negative numbers. But natural numbers may or may not contain 0, depending on the denomination. This name is misleading.

There is a word for natural numbers that contain zero `non-negative integers`.
It sounds good. But it is a redundant name.
ref: https://en.wikipedia.org/wiki/Natural_number

What about `Numerical digit`?
Numerical digit seems to be suitable because it means the symbols 0-9.
It is not as ostentatious as `non-negative integers`, so `numerical_digits` looks good!
ref: https://en.wikipedia.org/wiki/Numerical_digit